src/core/createEventManager.js

```diff
@@ -58,12 +58,28 @@ export default ({
           return consent.awaitConsent();
         })
         .then(() => {
-          // NOTE: this calls onBeforeEventSend callback (if configured)
-          event.finalize(onBeforeEventSend);
+          let shouldSend;
+          try {
+            // NOTE: this calls onBeforeEventSend callback (if configured)
+            event.finalize(onBeforeEventSend);
+            shouldSend = event.shouldSend();
```

Comment:
This line doesn't need to be part of the of `try...catch`, so let's pull it out and then down where you have `if (!shouldSend) {` we can do `if (!event.shouldSend()) {`